backend/de.metas.business/src/main/java/org/adempiere/mm/attributes/api/impl/AttributesBL.java

```diff
@@ -256,6 +265,19 @@ public ImmutableList<I_M_Attribute> getAttributesMandatoryOnPicking(final Produc
 		return attributesMandatoryOnPicking;
 	}
 
+	@Override
+	public ImmutableList<I_M_Attribute> getAttributesMandatoryOnManufacturing(final ProductId productId)
+	{
+		final AttributeSetId attributeSetId = productBL.getAttributeSetId(productId);
+		final ImmutableList<I_M_Attribute> attributesMandatoryOnManufacturing = attributesRepo.getAttributesByAttributeSetId(attributeSetId).stream()
+				.filter(attribute -> isMandatoryOn(productId,
```

Comment:
this feels like SQL N+1 issue... but i see other methods are doing the same.... so leave it like it is for now.